### PR TITLE
Lavaland

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
I gave up on river generation for now. Too many problems. I'm going to map some rivers of my own, and then it's ready.

:cl: Kor (And all the lovely maintainers and spriters who helped me along the way)
rscadd: Adds a new, lava themed planet to mine on,with randomly generated rivers of lava and islands of volcanic rock. This mine is used on Box and Metastation. Other mappers may enable it at their discretion.
rscadd: Added deadly ash storms to mining.
rscadd: Added slightly deadlier variants of mining mobs to lavaland, with sprites done by my girlfriend.
rscadd: Added portable survival capsules to mining equipment lockers. Don't forget these, or you might get caught out by a storm. You may purchase more at the venders.
rscadd: Added randomly loaded ruins to mining. These maps contain new and unique items not seen elsewhere in thegame. There are around a dozen of them added with the "release" of lavaland, though I plan for many more.
rscadd: Some ruins contain sleepers which allow you to spawn as various roles, such as survivors of a shuttle crash stranded in the wilderness. You can find these sleepers in the orbit list.
rscadd: You can now fly the abandoned white ship to lavaland.
/:cl:

Thank you @RemieRichards @WJohn  @tkdrg @duncathan @tkdrg @xxalpha @AnturK (and anyone else I forgot, but I think those were the big ones) for the massive amount of support, this would have died without your patience and help (and especially the patience, I know how infuriating my code must be to review at times).

### CODE REVIEW

**Comment by GunHog** [comment]
Kor ruins the codebase!

**Comment by Zilenan91** [comment]
Is this the goon thing or something else because I see lavaland in there

**Comment by duncathan** [comment]
interesting

**Comment by Shadowlight213** [comment]
Kor attempting to add fun. Close poste haste tbh fam

File: code/modules/mining/mine_items.dm [hunk 1]
```diff
@@ -186,6 +187,13 @@
 		src.loc.visible_message("The [src] begins to shake. Stand back!")
 		used = TRUE
 		sleep(50)
+		var/clear = TRUE
+		for(var/turf/T in range(src, 2))
```
Comment (RemieRichards) [inline code/modules/mining/mine_items.dm]: use ```trange()``` and then make this loop istypeless
repeat elsewhere if necessary

**Comment by Durkel** [comment]
Couple suggestions.
1. Get rid of the wild west maze, that maze is bad and you should feel bad.
2. Artifact rooms can spawn right outside the outpost, literaly the first thing you'll see it.
3. Max upgraded hardsuits with goliath plates can withstand ash storms. That's eight(ten?) goliath kills and I think that's a fair trade.

**Comment by Zilenan91** [comment]
10 goliath plates. Would be nice if that protected them yeah.

**Comment by Zilenan91** [comment]
Also Pls add the Abandoned Mining Station back in, it was a cool building that I'd like to see back.

**Comment by Zilenan91** [comment]
Hopefully last post, but mining mobs fight the Faithless inside of the Wild West vault at round start

**Comment by Boredone** [comment]
Based on what I was told(And double checking the code), you can become immune to the Ash Storms by using Dark Blue Slime Extracts for Chill potions, or maybe grabbing a CE/Atmos Hardsuit or an Atmos Firesuit.

**Comment by KorPhaeron** [comment]
@duncathan 

This is more or less ready for code review I think, though the mining turf shit is still hacky because mining turfs themselves are ugly and hacky. I just need to map some rivers and I'm done.

**Comment by duncathan** [comment]
don't even try to tell me mining turfs are shit kor
i have stared into the abyss of turf code and learned things no man was meant to learn

But yea I'll see about reviewing this in a little while.

**Comment by KorPhaeron** [comment]
So I fixed the source of roundstart active turfs, but opening a canister or breaking a survival shelter or something will still travel a long way down rivers/tunnels etc.

Maybe lavaland turfs should block air? Or all share one gas mixture like you're planning for space turfs @duncathan ?

**Comment by KorPhaeron** [comment]
@duncathan 

I think I've killed the worst bugs, just waiting on a couple fixes from @xxalpha to be merged first

**Comment by KorPhaeron** [comment]
Alright @RemieRichards @Cheridan @ChangelingRain @duncathan I think it's "done"

File: code/modules/mining/mine_turfs.dm [hunk 2]
```diff
@@ -97,59 +100,34 @@ var/global/list/rockTurfEdgeCache
...
+		var/turf/T = new path(src)
+
+		if(T && istype(T, /turf/simulated/mineral))
+			var/turf/simulated/mineral/M = T
+			M.mineralAmt = rand(1, 5)
+			M.environment_type = src.environment_type
+			M.turf_type = src.turf_type
+			M.baseturf = src.baseturf
+			src = M
+			M.levelupdate()
```
Comment (Cheridan) [inline code/modules/mining/mine_turfs.dm]: good code cleanup here :)

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm [hunk 3]
```diff
@@ -611,3 +611,128 @@
...
+
+/mob/living/simple_animal/hostile/asteroid/goliath/beast
+	name = "goliath"
+	desc = "A massive beast that uses long tentacles to ensare its prey, threatening them is not advised under any conditions."
+	icon = 'icons/mob/lavaland/lavaland_monsters.dmi'
+	icon_state = "goliath"
+	icon_living = "goliath"
+	icon_aggro = "goliath"
+	icon_dead = "goliath_dead"
+	throw_message = "does nothing to the touch hide of the"
```
Comment (Cheridan) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: tough hide*

**Comment by KorPhaeron** [comment]
@Cheridan 

Okay I tested more locally it's ready for real

**Comment by KorPhaeron** [comment]
Glad that is over with

**Pull request merged**